classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -156,7 +141,18 @@ public function fetch_action( $action_id ) {
 		}
 
 		if ( ! empty( $data->extended_args ) ) {
-			$data->args = $data->extended_args;
+			$extended_args = json_decode( $data->extended_args, true );
+			// Stored data from before the extended usage
+			if ( ! is_array( $extended_args ) ) {
+				$data->args = $data->extended_args;
+				$extended_args = array();
+			}
+			if ( array_key_exists( 'args', $extended_args ) ) {
+				$data->args = $extended_args['args'];
```

Comment:
@n3f I know you haven't marked this PR for review yet. This implementation of retry storage looks like it might be problematic in a couple ways:

- Prior to the change in this block of code `$data->args` is a `json_encoded` array. With this block it's an array. In `make_action_from_db_record()` it is passed to `json_decode()` which expects a string
- The action search includes the extended args field https://github.com/woocommerce/action-scheduler/blob/version_3_2_0/classes/data-stores/ActionScheduler_DBStore.php#L346 and adding non-args data to that field could result in false positives

In addition, there are other use cases where retries would be useful (eg. Subscriptions). Adding a new field to the schema would simplify this PR and leave room for other implementations.